client/src/helper/utils.ts

```diff
@@ -1,76 +1,81 @@
 import { isString, isInteger } from "lodash-es";
 import { utcToZonedTime } from "date-fns-tz";
 
-export function isIsoDateWithoutTime(string) {
+export function isIsoDateWithoutTime(string: string) {
   const regex = /^\d{4}-(0[0-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1])$/g;
   return string?.match(regex) != null;
 }
 
-export function setWindowName(namePostfix) {
+export function setWindowName(namePostfix: string) {
   document.title = `${import.meta.env.VITE_PAGE_TITLE_PREFIX}${namePostfix}`;
 }
-export function retrieveDateOnly(isoDate) {
+export function retrieveDateOnly(isoDate?: string) {
+  if (!isoDate) return "";
   return isoDate.substring(0, 10);
 }
-export function dayAfter(date) {
+export function dayAfter(date?: string) {
+  if (!date) return "";
   const dateObject = new Date(date);
   dateObject.setDate(dateObject.getDate() + 1);
   return retrieveDateOnly(dateObject.toISOString());
 }
 
-export function adjustDateToLocal(originalDate) {
+export function adjustDateToLocal(originalDate: string) {
   const tz = import.meta.env.VITE_BASE_TZ || "Europe/Berlin";
   return utcToZonedTime(new Date(originalDate).getTime(), tz);
 }
 
-export function isEmail(value) {
-  if (!isString(value)) return;
+export function isEmail(value: string) {
+  if (!isString(value)) return; // TODO: Can be removed if all files use TS
   return value.match(/^[^\s@]+@[^\s@]+\.[^\s@]+$/);
 }
 
-export function isInt(value) {
+export function isInt(value: string) {
   return isInteger(parseInt(value));
 }
-
-export function findKeyByValue(object, value) {
+// TODO: How to do this in properly in TS?
+export function findKeyByValue<T1, T2>(object: T1, value: T2) {
+  // @ts-ignore
   return Object.keys(object).find((k) => object[k] == value);
 }
 
-export function isCoordinate(value) {
-  if (!isString(value)) return;
+export function isCoordinate(value: string) {
+  if (!isString(value)) return; // Can be removed if all files use TS
   return value.match(/^-?\d{0,3}.\d{4,16}$/);
 }
 
-export function isDirection(value) {
-  if (!isString(value)) return;
+export function isDirection(value: string) {
+  if (!isString(value)) return; // Can be removed if all files use TS
   return value.match(/^[NSOWnsow]{1,3}[-/,]?[NSOWnsow]{0,3}$/);
 }
 
-export function isStrongPassword(value) {
-  if (!isString(value)) return;
+export function isStrongPassword(value: string) {
+  if (!isString(value)) return; // Can be removed if all files use TS
   const regex =
     /^(?=(.*[a-z]){1,})(?=(.*[A-Z]){1,})(?=(.*[0-9]){1,})(?=(.*[!@#$%/=?^&*()<>\-__+.]){1,}).{8,}$/;
   return value.match(regex);
 }
 
-export async function asyncForEach(array, callback) {
+export async function asyncForEach<T1>(
```

Comment:
Wenn du das await entfernst, dann brauchst du auch das async nicht mehr.
Und dann brauchst du auch eigentlich die ganze Funktion nicht mehr.

Reply:
doch, denn da kommt ja ne async Funktion in den cb.